core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1556,6 +1681,133 @@ public void warmUp() {
+                for (int i = 0, limit = detachedMetadata.getColumnCount(); i < limit; i++) {
+                    String columnName = detachedMetadata.getColumnName(i);
+                    if (metadata.getColumnIndexQuiet(columnName) == -1) {
+                        ff.walk(detachedPath, (pUtf8NameZ, type) -> {
+                            if (type == Files.DT_FILE) {
+                                fileNameSink.clear();
+                                Chars.utf8DecodeZ(pUtf8NameZ, fileNameSink);
+                                int len = fileNameSink.length();
+                                int dotIdx = len - 1;
+                                while (dotIdx > 0 && fileNameSink.charAt(dotIdx) != '.') {
+                                    dotIdx--;
+                                }
+                                if (dotIdx > 0) {
+                                    CharSequence cn = fileNameSink.subSequence(0, dotIdx);
+                                    if (Chars.equals(cn, columnName)) {
+                                        other2.parent().concat(fileNameSink).$();
+                                        if (!ff.remove(other2)) {
+                                            LOG.info().$("failed to remove [path=").$(other2).I$();
+                                        }
+                                    }
+                                }
+                            }
+                        });
+                    }
+                }
+            }
+        } finally {
+            Misc.free(detachedMetadata);
+            // remove _meta
+            if (!ff.remove(other2)) {
+                LOG.error().$("cannot remove [errno=").$(ff.errno())
+                        .$(", path=").$(other2)
+                        .I$();
+            }
+        }
+
+        // load/check _cv, updating local column tops
+        // set current _cv to where the partition was
+        other2.parent().concat(COLUMN_VERSION_FILE_NAME).$();
+        if (!ff.exists(other2)) {
+            // Backups and older versions of detached partitions will not have _cv
+            LOG.info().$("detached _cv not found, skipping check [path=").$(other2).I$();
+            return;
+        }
+        try {
+            if (detachedColumnVersionReader == null) {
+                detachedColumnVersionReader = new ColumnVersionReader();
+            }
+            detachedColumnVersionReader.ofRO(ff, other2);
+            detachedColumnVersionReader.readSafe(MillisecondClockImpl.INSTANCE, Long.MAX_VALUE);
+            LongList entries = columnVersionWriter.getCachedList();
+            LongList detachedEntries = detachedColumnVersionReader.getCachedList();
+            for (int i = 0, limit = detachedEntries.size(); i < limit; i += 4) {
+                // detachedEntries: [partitionTimestamp, columnIndex, columnNameTxn, columnTop]*
+                long columnTop = detachedEntries.getQuick(i + 3);
```

Comment:
Column indexes may not match if it's different table. We either should not allow to have column indexes mismatch or handle this here.

Reply:
resolved.